docs/assembly.md

```diff
@@ -0,0 +1,184 @@
+`methods`    |`Method[]`   |The [methods] declared by this interface
+`name`       |`string`     |The simple name of the interface
+`properties` |`Property[]` |The [properties] declared by this interface
+
+#### Structures (a.k.a. Data Types)
+
+*Structures* (or *Data Types*) are immutable, data-only interfaces:
+* They declare no methods
+* All [properties] they declare are `readonly`
+* They can only implement other *structures*
+* They cannot be extended by interfaces that are not *structures*
+* They cannot be implemented by *classes*
+
+Unlike regular *interfaes*, `jsii` *structure* names are not required to have
+any particular prefix.
+
+Since those are immutable, pure data objects, the `jsii-runtime` exchanges
+instances of those *by value*, instead of *by reference*, allowing to save
+cross-language communication overhead when working with the data.
+
+Attribute    | Type        | Description
+-------------|-------------|----------------------------------------------------
+`kind`       |`'inteface'` |Discriminator to identify interfaces
+`datatype`   |`true`       |Indicates a *structure* / *data type* declaration
+`assembly`   |`string`     |The name of the assembly this structure is a part of
+`fqn`        |`string`     |The fully-qualified name of the structure
+`interfaces` |`string[]`   |The fully-qualified names of *structures* extended by this *structure*
+`name`       |`string`     |The simple name of the structure
+`properties` |`Property[]` |The [properties] declared by this structure (all `readonly`)
+
+#### Enums
+
+Attribute    | Type         | Description
+-------------|--------------|---------------------------------------------------
+`kind`       |`'enum'`      |Discriminator to identify enums
+`assembly`   |`string`      |The name of the assembly this enum is a part of
+`fqn`        |`string`      |The fully-qualified name of the enum
+`members`    |`EnumMember[]`|The [enum members] declared by this enum
+`name`       |`string`      |The simple name of the enum
+
+[enum members]: #enum-members
+
+### Members
+
+#### Constructors
+
+Attribute    | Type         | Description
+-------------|--------------|---------------------------------------------------
+`overrides`  |`string`      |The fully-qualified name of the class/interface that declares the overridden constructor
+`parameters` |`Parameter[]` |Parameters of this constructor
+`protected`  |`boolean`     |Whether this constructor is protected
+`variadic`   |`boolean`     |Whether the last parameter is `variadic`
+
+#### Enum Members
+
+Attribute    | Type         | Description
+-------------|--------------|---------------------------------------------------
+`name`       |`string`      |The name of the enum member. Must be `UPPER_SNAKE_CASED`
+
+#### Methods
+
+Attribute    | Type          | Description
+-------------|---------------|--------------------------------------------------
+`abstract`   |`boolean`      |Whether this method is `abstract`
```

Comment:
I think the descriptions are a bit lacking. This is where we need to define what each attribute means in terms of design and modeling. For example, `abstract`..

Reply:
I know. I wanna add more detailed verbiage, this is intended to be a high-level (birds-eye) view. I just want to ship this now - it's better than nothing.